### DISCUSSION TITLE
Callout boxes for markdown output

### DISCUSSION THREAD

**Discussion by linogaliana** [quarto-dev/quarto-cli#1495] [Feature Requests]
Hi !

- In the doc about `callouts` (https://quarto.org/docs/authoring/callouts.html), `markdown` is not part of the supported format. Since it is possible to embed html blocks within `.md`, would it be possible to add `markdown` to the supported format ? 

- I suppose this means adding something based on `_quarto.format.isMarkdownOutput()` to the list of supported formats :

https://github.com/quarto-dev/quarto-cli/blob/852c9bb99e54e0163575ceecf7dde372b4b73321/src/resources/filters/quarto-pre/callout.lua#L60-L61

- I guess this should be far easier than #1167 since in my mind this could really be the same output than the HTML version (`return calloutDiv(div)`). 

- Please tell me if I don't see some drawbacks (I am unfamiliar with `lua` so might miss some difficulties)

**Comment by cderv** [comment]
> Since it is possible to embed html blocks within .md

Callouts for HTML requires requires Bootstrap for styling. With markdown outputs, you can't really add styling for the callouts. Often styling will depend of the previewer of renderer of Markdown to HTML. Callouts in GFM won't work for example. 

How do you see things regarding inserting HTML blocks requiring an external lib like bootstrap in a markdown file ? 
I am wondering the use case that lead to this question.

**Reply by linogaliana** [reply]
Hi Christophe !

In my Python website that is based on `quarto` (qmd -> md) and `hugo` (md -> html) , I use wowchemy theme. Two years ago, I added a [Hugo shortcode](https://github.com/linogaliana/python-datascientist/blob/master/layouts/shortcodes/box.html) to generate callouts. 

For instance, 

https://github.com/linogaliana/python-datascientist/blob/2efce7829f717055932b5ce4966de328b07b38c6/content/course/manipulation/02a_pandas_tutorial/index.qmd#L80-L116

generates this box in https://pythonds.linogaliana.fr/pandas/ :

![image](https://user-images.githubusercontent.com/33896139/180448347-d4bad841-8969-4550-9dd4-bb14d83cd9f8.png)

Using a custom hugo shortcode works fine now but is not really clever in terms of maintenance and is hugo relevant (while one of the great advantage of `quarto` in my opinion is the way it handles with the same source code multiple output). Since `quarto` callouts seem to follow exactly the same principle (given minimal change in the structure of the callout arguments), I was thinking that my custom shortcode reinvents the wheel. 

Regarding your question, I suppose that my hugo theme handles bootstrap. I suppose this is the case of most hugo themes, am I right ?

**Comment by Aariq** [comment]
Github-flavored markdown seems to support callouts like 

```markdown
>[!NOTE]
> a note

>[!WARNING]
>  a warning

>[!IMPORTANT]
> something important
```
>[!NOTE]
> a note

>[!WARNING]
>  a warning

>[!IMPORTANT]
> something important

It would be nice if the corresponding callouts in Quarto were translated to this upon rendering to `gfm`

**Reply by cscheid** [reply]
That's a good idea, and shouldn't be too hard for us to do since 1.3. Let's open a new issue to track

**Reply by cscheid** [reply]
Tracking here: https://github.com/quarto-dev/quarto-cli/issues/6935

**Reply by cderv** [reply]
Really cool ! I did not know that. Thanks for sharing ! 

It seems it is still in Beta so it would probably be worth waiting for stability
* https://github.com/orgs/community/discussions/16925

**Reply by Aariq** [reply]
I can't take full credit for this—I found out about this feature from Don McCurdy on the fediverse: https://fosstodon.org/@donmccurdy/111103471765920844

**Comment by mcanouil** [comment]
Closing in favour of:

- #6935
